engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/AnimationScreen.java

```diff
@@ -0,0 +1,127 @@
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.characters.StandComponent;
+import org.terasology.logic.characters.WalkComponent;
+import org.terasology.logic.common.InspectionToolComponent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.rendering.nui.BaseInteractionScreen;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.widgets.*;
+
+import java.util.ArrayList;
+
+/**
+ */
+public class AnimationScreen extends CoreScreenLayer {
+    private UIText fullDescriptionLabel;
+    private UIText entityIdField;
+    private UIButton spawnEntityIdButton;
+    private UISlider animationSpeedSlider;
+    @In
+    private EntityRef entityRef;
+    @In
+    private LocalPlayer localPlayer;
+    @In
+    private EntityManager entityManager;
+    @In
+    private AssetManager assetManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(AnimationScreen.class);
+
+
+    @Override
+    public void initialise() {
+        spawnEntityIdButton = find("spawnEntityIdButton", UIButton.class);
+        UIDropdown<ResourceUrn> entityDropdown = find("entityDropdown", UIDropdownScrollable.class);
+        logger.info("Number of available skeletal meshes: " + assetManager.getAvailableAssets(SkeletalMesh.class).size());
+        if (entityDropdown != null) {
+            entityDropdown.setOptions(new ArrayList(assetManager.getAvailableAssets(SkeletalMesh.class)));
+        }
+        animationSpeedSlider = find("entityAnimationSpeedSlider", UISlider.class);
+        if (animationSpeedSlider != null) {
+            animationSpeedSlider.setMinimum(-2.5f);
+            animationSpeedSlider.setIncrement(0.1f);
+            animationSpeedSlider.setRange(5.0f);
+            animationSpeedSlider.setPrecision(1);
+        }
+        spawnEntityIdButton.subscribe(widget -> {
+            Vector3f localPlayerPosition = localPlayer.getPosition();
+            Quat4f localPlayerRotation = localPlayer.getRotation();
+            java.util.Optional<Prefab> prefab = assetManager.getAsset(entityDropdown.getSelection(), Prefab.class);
+            if (prefab.isPresent() && prefab.get().getComponent(LocationComponent.class) != null)
+            entityRef = entityManager.create(prefab.get(), localPlayerPosition, localPlayerRotation);
+            SkeletalMeshComponent skeletalMeshComponent = entityRef.getComponent(SkeletalMeshComponent.class);
+            skeletalMeshComponent.animationRate = animationSpeedSlider.getValue();
```

Comment:
You need some validation code here. Right now the slider offers negative values, yet if you pick one the game crashes:

```
java.lang.IndexOutOfBoundsException: index (-1) must not be negative
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:310)
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:293)
	at com.google.common.collect.RegularImmutableList.get(RegularImmutableList.java:67)
	at org.terasology.rendering.assets.animation.MeshAnimationImpl.getFrame(MeshAnimationImpl.java:64)
	at org.terasology.rendering.logic.SkeletonRenderer.updateSkeletalMeshOfEntity(SkeletonRenderer.java:169)
```